core/shared/src/main/scala/zio/Queue.scala

```diff
@@ -248,26 +244,35 @@ object Queue extends QueuePlatformSpecific {
     override def isShutdown(implicit trace: Trace): UIO[Boolean] = ZIO.succeed(shutdownFlag.get)
 
     override def take(implicit trace: Trace): UIO[A] =
-      ZIO.fiberIdWith { fiberId =>
-        if (shutdownFlag.get) ZIO.interrupt
-        else {
-          queue.poll(null.asInstanceOf[A]) match {
-            case null =>
-              // add the promise to takers, then:
-              // - try take again in case a value was added since
-              // - wait for the promise to be completed
-              // - clean up resources in case of interruption
-              val p = Promise.unsafe.make[Nothing, A](fiberId)(Unsafe.unsafe)
-
-              ZIO.suspendSucceed {
+      ZIO.uninterruptibleMask { restore =>
+        ZIO.fiberIdWith { fiberId =>
+          if (shutdownFlag.get) ZIO.interrupt
```

Comment:
What does `ZIO.interrupt` do when wrapped in a `ZIO.uninterruptibleMask`? 🤔 
Do we need to `restore` it?